span-processing-config-service-impl/src/main/java/org/hypertrace/span/processing/config/service/SpanProcessingConfigServiceImpl.java

```diff
@@ -67,10 +104,35 @@ public void getAllExcludeSpanRules(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       this.validator.validateOrThrow(requestContext, request);
+      List<ExcludeSpanRuleDetails> excludeSpanRules =
+          excludeSpanRulesConfigStore.getAllData(requestContext);
+
+      Map<String, ExcludeSpanRuleDetails> tenantLevelExcludeSpanRulesToIdMap =
+          buildTenantLevelExcludeSpanRulesToIdMap(excludeSpanRules);
+
+      // all the tenant level configs except for the overridden system level ones
+      List<ExcludeSpanRuleDetails> filteredTenantLevelExcludeSpanRules =
+          excludeSpanRules.stream()
+              .filter(
+                  excludeSpanRule ->
+                      !systemLevelExcludeSpanRulesToIdMap.containsKey(
+                          excludeSpanRule.getRule().getId()))
+              .collect(Collectors.toUnmodifiableList());
+
+      // all the system level configs, replaced by the overridden configs in case of overrides
+      List<ExcludeSpanRuleDetails> filteredSystemLevelExcludeSpanRules =
+          systemLevelExcludeSpanRules.stream()
+              .map(
+                  excludeSpanRule ->
+                      tenantLevelExcludeSpanRulesToIdMap.containsKey(excludeSpanRule.getId())
+                          ? tenantLevelExcludeSpanRulesToIdMap.get(excludeSpanRule.getId())
+                          : ExcludeSpanRuleDetails.newBuilder().setRule(excludeSpanRule).build())
+              .collect(Collectors.toUnmodifiableList());
 
       responseObserver.onNext(
           GetAllExcludeSpanRulesResponse.newBuilder()
-              .addAllRuleDetails(excludeSpanRulesConfigStore.getAllData(requestContext))
+              .addAllRuleDetails(filteredTenantLevelExcludeSpanRules)
+              .addAllRuleDetails(filteredSystemLevelExcludeSpanRules)
```

Comment:
This looks complex. I don't think there was a need to filter out overrides from persisted exclude span rules. We could have actually just sent out the entire persisted exclude span rules + non overriden system exclude rules

Something like this

```
Set<String> userExcludeSpanRuleIds = excludeSpanRules.stream()
    .map(excludeSpanRule -> excludeSpanRule.getRule().getId()).collect(
        Collectors.toUnmodifiableSet());

// all the system level configs, replaced by the overridden configs in case of overrides
List<ExcludeSpanRule> nonOverridenExcludeSpanRules =
    systemLevelExcludeSpanRules.stream()
        .filter(
            systemExcludeRule -> !userExcludeSpanRuleIds.contains(systemExcludeRule.getId()))
        .collect(Collectors.toUnmodifiableList());
```

And, we can then convert the `nonOverridenExcludeSpanRules` to `List<ExcludeSpanRuleDetails>` while sending it in the response

Reply:
I had taken this route so as to ensure that the system exclude rules(overridden or not) would end up in the bottom and the UI wouldn't have to worry about the ordering.